packages/stripe-integration/src/stripe-upe/stripe-upe-integration-service.ts

```diff
@@ -0,0 +1,222 @@
+import { includes, some } from 'lodash';
+
+import {
+    Address,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedError,
+    NotInitializedErrorType,
+    PaymentIntegrationService,
+    PaymentMethodFailedError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import {
+    AddressOptions,
+    StripeAdditionalActionRequired,
+    StripeConfirmPaymentData,
+    StripeElement,
+    StripeElements,
+    StripeElementType,
+    StripeError,
+    StripeStringConstants,
+    StripeUPEClient,
+    StripeUPEPaymentIntentStatus,
+} from './stripe-upe';
+import StripeUPEPaymentInitializeOptions from './stripe-upe-initialize-options';
+
+export default class StripeUPEIntegrationService {
+    private isMounted = false;
+    private checkoutEventsUnsubscribe?: () => void;
+
+    constructor(private paymentIntegrationService: PaymentIntegrationService) {}
+
+    deinitialize(): void {
+        this.checkoutEventsUnsubscribe?.();
+        this.isMounted = false;
+    }
+
+    initCheckoutEventsSubscription(
+        gatewayId: string,
+        methodId: string,
+        stripeupe: StripeUPEPaymentInitializeOptions,
+        stripeElements?: StripeElements,
+    ): void {
+        this.checkoutEventsUnsubscribe = this.paymentIntegrationService.subscribe(
+            async () => {
+                const paymentElement = stripeElements?.getElement(StripeElementType.PAYMENT);
+
+                if (!paymentElement) {
+                    return;
+                }
+
+                let error;
+
+                try {
+                    await this.paymentIntegrationService.loadPaymentMethod(gatewayId, {
```

Comment:
Should payment method be loaded each time when `initCheckoutEventsSubscription` is triggered? Can we check if payment method is available in checkout state (using getPaymentMethod), and load it only if it does not exist?

Reply:
This config loading uses for triggering Payment Intent update on backend side. To sync actual state of payment data between BigPay and Stripe side. So we need to call it each time when price or some other data is changed.

Reply:
Should we leave a comment then so other developers do not remove this by mistake (or refactoring)?

Reply:
Add new method with a comment? to make this logic more transparent
[https://github.com/bigcommerce/checkout-sdk-js/pull/2710/commits/bffde45e9422f860b94448c581bbad816d78f3fb](https://github.com/bigcommerce/checkout-sdk-js/pull/2710/commits/bffde45e9422f860b94448c581bbad816d78f3fb)